### PR TITLE
Adding recommended storage guidelines to scaling and performance guide

### PR DESCRIPTION
**Pull request by ghost**
Related to https://github.com/openshift/openshift-docs/pull/5623

### CODE REVIEW

**Comment by tmorriso-rh** [comment]
@lpsantil Hi Louis, this is the follow up PR for https://github.com/openshift/openshift-docs/pull/5623. Please review the files changed. Thanks.

**Comment by lpsantil** [comment]
LGTM

**Comment by lpsantil** [comment]
LGTM

**Comment by tmorriso-rh** [comment]
[rev_history]
|xref:../scaling_performance/optimizing_storage.adoc[Optimizing Storage]
|Added the xref:../scaling_performance/optimizing_storage.adoc#general-storage-guidelines[General Storage Guidelines] section for optimizing OpenShift storage.
%

**Comment by tmorriso-rh** [comment]
@openshift/team-documentation Please peer review this PR and revision history.

File: scaling_performance/optimizing_storage.adoc
Comment (gaurav-nelson) [inline scaling_performance/optimizing_storage.adoc]: s/OpenShift/{product-title}

File: scaling_performance/optimizing_storage.adoc
Comment (gaurav-nelson) [inline scaling_performance/optimizing_storage.adoc]: We can set column widths to `[cols="1,4,3",options="header"]` instead, as it renders better.

File: scaling_performance/optimizing_storage.adoc
Comment (gaurav-nelson) [inline scaling_performance/optimizing_storage.adoc]: maybe s/capabilities will vary widely/capabilities varies widely

File: scaling_performance/optimizing_storage.adoc
Comment (gaurav-nelson) [inline scaling_performance/optimizing_storage.adoc]: s/via/through

File: scaling_performance/optimizing_storage.adoc
Comment (gaurav-nelson) [inline scaling_performance/optimizing_storage.adoc]: Usually I prefer `✅` for Yes and `🞩` for No. Up to you if you want to change that. :innocent:

File: scaling_performance/optimizing_storage.adoc
Comment (gaurav-nelson) [inline scaling_performance/optimizing_storage.adoc]: Maybe we should explain the full form of ROX and RWX before we use them?

File: scaling_performance/optimizing_storage.adoc
Comment (gaurav-nelson) [inline scaling_performance/optimizing_storage.adoc]: For this section, I think we should use points for each sentence, like:
```
In a non-scaled/HA {product-title} registry cluster deployment:

* The preferred storage back-end type is object storage followed by block
storage.  The storage back-end type does not need to support ReadWriteMany
* (RWX) access mode.  The storage back-end type must ensure read-after-write
* consistency.  All NAS back-end types (excluding CNS/CRS GlusterFS as it uses
an object storage interface) are expressly not recommended for {product-title}
Registry cluster deployment with production workloads.  
* While hostPath volumes are configurable for a non-scaled/HA {product-title}
Registry cluster deployment, they are not recommended.
```

Comment (gaurav-nelson) [reply scaling_performance/optimizing_storage.adoc]: Same for Scaled registry, metrics, logging, and applications section if possible.

Comment (gaurav-nelson) [reply scaling_performance/optimizing_storage.adoc]: Should we also highlight whats not recommend by using admonitions? I am not sure. :confused: :grey_question:

File: scaling_performance/optimizing_storage.adoc
Comment (gaurav-nelson) [inline scaling_performance/optimizing_storage.adoc]: `Btrfs` is the correct term https://en.wikipedia.org/wiki/Btrfs

File: scaling_performance/optimizing_storage.adoc
Comment (gaurav-nelson) [inline scaling_performance/optimizing_storage.adoc]: Portable Operating System Interface (POSIX)

**Comment by gaurav-nelson** [comment]
Some comments and questions form me @tmorriso-rh rest LGTM :chart_with_upwards_trend:

File: scaling_performance/optimizing_storage.adoc
Comment (lpsantil) [reply scaling_performance/optimizing_storage.adoc]: @gaurav-nelson YES!  We do need to mention what to use. Here is an email from openshift-users email list from yesterday, 11/1/2017.

```
I am confused over persistent storage for logging (elasticsearch).

The latest advanced installer docs [1] specifically describes how to define using NFS for persistent storage, but the docs for "aggregating container logs" [2] says that NFS should not be used (except in one particular scenario) and seems to suggest that the only really suitable scenario is to use a volume (disk) directly mounted to each logging node.

Could someone clarify the situtation?
```

My goal is to reference this page from the Registry, Metrics, and Logging install docs once this PR lands.  There are probably a couple other places which could do with a reference to this doc as well.

File: scaling_performance/optimizing_storage.adoc
Comment (nengard) [inline scaling_performance/optimizing_storage.adoc]: delete 'sometimes'

File: scaling_performance/optimizing_storage.adoc
Comment (nengard) [inline scaling_performance/optimizing_storage.adoc]: delete 'sometimes'

File: scaling_performance/optimizing_storage.adoc
Comment (nengard) [inline scaling_performance/optimizing_storage.adoc]: "storage back-end types" is very clunky.  IBM styleguide does suggest using something more detailed than 'back-end' if possible and 'types' I also feel need rewording.

Would 'storage systems' make just as much sense? Or does that change the meaning?

Comment (nengard) [reply scaling_performance/optimizing_storage.adoc]: This goes for all occurrences of this phrase that follow.

File: scaling_performance/optimizing_storage.adoc
Comment (nengard) [inline scaling_performance/optimizing_storage.adoc]: We should be sure to spell out HA the first time it appears in a chapter.

File: scaling_performance/optimizing_storage.adoc
Comment (nengard) [inline scaling_performance/optimizing_storage.adoc]: Should 'hostPath' be styled as a literal?

File: scaling_performance/optimizing_storage.adoc
Comment (nengard) [inline scaling_performance/optimizing_storage.adoc]: back-end has a hyphen - but if you can avoid using the word that's best.

File: scaling_performance/optimizing_storage.adoc
Comment (nengard) [inline scaling_performance/optimizing_storage.adoc]: Is containers supposed to be possessive here?

File: scaling_performance/optimizing_storage.adoc
Comment (nengard) [inline scaling_performance/optimizing_storage.adoc]: Shouldn't 'overlay' and 'device mapper' be styled as literals?

File: scaling_performance/optimizing_storage.adoc
Comment (nengard) [inline scaling_performance/optimizing_storage.adoc]: I think this should be 'In an {product-title}...'

File: scaling_performance/optimizing_storage.adoc
Comment (nengard) [inline scaling_performance/optimizing_storage.adoc]: replace 'via' with 'from'

File: scaling_performance/optimizing_storage.adoc
Comment (nengard) [inline scaling_performance/optimizing_storage.adoc]: I think this should be 'In an {product-title}...'

File: scaling_performance/optimizing_storage.adoc
Comment (nengard) [inline scaling_performance/optimizing_storage.adoc]: replace 'via' with 'from'

File: scaling_performance/optimizing_storage.adoc
Comment (nengard) [inline scaling_performance/optimizing_storage.adoc]: Should 'hosted logging' be capitalized? It's not in the previous section. Maybe it should be treated as a 'literal' in both places?

File: scaling_performance/optimizing_storage.adoc
Comment (nengard) [inline scaling_performance/optimizing_storage.adoc]: Issue 'with' production workloads.

File: scaling_performance/optimizing_storage.adoc
Comment (nengard) [inline scaling_performance/optimizing_storage.adoc]: Define 'PV' the first time you use it.

Comment (nengard) [inline scaling_performance/optimizing_storage.adoc]: Does "OpenShift Container Platform" have an attribute?

File: scaling_performance/optimizing_storage.adoc
Comment (nengard) [inline scaling_performance/optimizing_storage.adoc]: comma after 'mechanisms'

File: scaling_performance/optimizing_storage.adoc
Comment (nengard) [reply scaling_performance/optimizing_storage.adoc]: If the latter then it's 'capabilities vary widely'

**Review by ncbaratta — CHANGES_REQUESTED** [review]
I need to do another review after the suggested changes are made.

File: scaling_performance/optimizing_storage.adoc
Comment (lpsantil) [reply scaling_performance/optimizing_storage.adoc]: @nengard I "storage systems" is too specific, IMO.  That would refer to a specific vendor or even vendor solution for vendors that offer more than one product/solution.

I think agree with you now.  Maybe, you're searching for "storage technology".

File: scaling_performance/optimizing_storage.adoc
Comment (lpsantil) [reply scaling_performance/optimizing_storage.adoc]: :+1:

Comment (lpsantil) [reply scaling_performance/optimizing_storage.adoc]: Styled as `Overlay` and `DeviceMapper`.

File: scaling_performance/optimizing_storage.adoc
Comment (lpsantil) [reply scaling_performance/optimizing_storage.adoc]: Yes.

File: scaling_performance/optimizing_storage.adoc
Comment (tmorriso-rh) [reply scaling_performance/optimizing_storage.adoc]: Yes.

**Comment by tmorriso-rh** [comment]
@nengard and @lpsantil Changes made. PTAL. Thanks.

File: scaling_performance/optimizing_storage.adoc
Comment (nengard) [inline scaling_performance/optimizing_storage.adoc]: Not sure we want to say 'topic'.

File: scaling_performance/optimizing_storage.adoc
Comment (nengard) [inline scaling_performance/optimizing_storage.adoc]: "The following table lists the available persistent storage options for {product-title}."

File: scaling_performance/optimizing_storage.adoc
Comment (nengard) [inline scaling_performance/optimizing_storage.adoc]: Is this necessary? "This type of storage is presented to the operating system as a block device."

Again, try to minimize the words. A table is supposed to be easy to browse through:

"Referred to as a Storage Area Network (SAN). This type of storage is suitable for applications that need full control of storage and operate at a low level on files bypassing the file system.  It is usually a non-shareable type of storage, which means that only one client at a time can mount an endpoint of this type."

Consider doing the same will the other descriptions below.

File: scaling_performance/optimizing_storage.adoc
Comment (nengard) [inline scaling_performance/optimizing_storage.adoc]: Are footnotes common in these books? If not I'd use Notes not Footnotes.  Footnotes are a very academic paper thing and not common in technical documentation.

File: scaling_performance/optimizing_storage.adoc
Comment (nengard) [inline scaling_performance/optimizing_storage.adoc]: Remove "s" from "provides"

File: scaling_performance/optimizing_storage.adoc
Comment (nengard) [inline scaling_performance/optimizing_storage.adoc]: spell out '3'

File: scaling_performance/optimizing_storage.adoc
Comment (nengard) [inline scaling_performance/optimizing_storage.adoc]: Is hostPath a literal? Or a word?

Comment (nengard) [inline scaling_performance/optimizing_storage.adoc]: Reword to finish the sentence '... cluster deployment: ...'

File: scaling_performance/optimizing_storage.adoc
Comment (nengard) [inline scaling_performance/optimizing_storage.adoc]: 'expressly' isn't needed.  It's 'not recommended' is strong enough.

File: scaling_performance/optimizing_storage.adoc
Comment (nengard) [inline scaling_performance/optimizing_storage.adoc]: This needs more - something like "the following are some examples" or some such lead into the list.

File: scaling_performance/optimizing_storage.adoc
Comment (nengard) [inline scaling_performance/optimizing_storage.adoc]: etcd was in bold earlier ...

File: scaling_performance/optimizing_storage.adoc
Comment (nengard) [inline scaling_performance/optimizing_storage.adoc]: what is ROX

File: scaling_performance/optimizing_storage.adoc
Comment (nengard) [inline scaling_performance/optimizing_storage.adoc]: why isn't Btrfs monospaced as well?

File: scaling_performance/optimizing_storage.adoc
Comment (nengard) [inline scaling_performance/optimizing_storage.adoc]: There was no hyphen in thin pool above - so either add up there, or remove here.

File: scaling_performance/optimizing_storage.adoc
Comment (nengard) [inline scaling_performance/optimizing_storage.adoc]: Add hyphen? or leave and remove above.

File: scaling_performance/optimizing_storage.adoc
Comment (nengard) [inline scaling_performance/optimizing_storage.adoc]: thin-pool or thin pool - be consistent.

**Review by ncbaratta — CHANGES_REQUESTED** [review]

File: scaling_performance/optimizing_storage.adoc
Comment (lpsantil) [inline scaling_performance/optimizing_storage.adoc]: s/all NAS/NAS/

File: scaling_performance/optimizing_storage.adoc
Comment (lpsantil) [reply scaling_performance/optimizing_storage.adoc]: Technology (or technologies) is a appropriate here.  Most of the technologies are only supported buy purchasing them from a 3rd party.

File: scaling_performance/optimizing_storage.adoc
Comment (lpsantil) [reply scaling_performance/optimizing_storage.adoc]: I was intentionally this verbose because not all OpenShift administrators will have strong Storage experience.  The verbosity here provides baseline understanding w/o resorting to citation heavy descriptions.

File: scaling_performance/optimizing_storage.adoc
Comment (lpsantil) [reply scaling_performance/optimizing_storage.adoc]: Literal.  It should be stylized as `hostPath`.

File: scaling_performance/optimizing_storage.adoc
Comment (lpsantil) [reply scaling_performance/optimizing_storage.adoc]: I chose the word 'expressly' since the current documentation is so weak on the topic [0][1][2].  I'd like to link to this document from those docs once this lands.


[0] https://docs.openshift.com/container-platform/3.6/install_config/registry/deploy_registry_existing_clusters.html#registry-production-use
[1] https://docs.openshift.com/container-platform/3.6/install_config/aggregate_logging.html#aggregated-elasticsearch
[2] https://docs.openshift.com/container-platform/3.6/install_config/cluster_metrics.html#capacity-planning-for-openshift-metrics

File: scaling_performance/optimizing_storage.adoc
Comment (lpsantil) [reply scaling_performance/optimizing_storage.adoc]: ReadOnlyMany [0][1].

[0] https://docs.openshift.com/container-platform/3.6/architecture/additional_concepts/storage.html#pv-access-modes
[1] https://kubernetes.io/docs/concepts/storage/persistent-volumes/#access-modes

File: scaling_performance/optimizing_storage.adoc
Comment (nengard) [reply scaling_performance/optimizing_storage.adoc]: Thanks @lpsantil so then:

"The following table lists the available persistent storage technologies for {product-title}."

File: scaling_performance/optimizing_storage.adoc
Comment (nengard) [reply scaling_performance/optimizing_storage.adoc]: I think we can still get across the same amount of information without so many extraneous words.

File: scaling_performance/optimizing_storage.adoc
Comment (nengard) [reply scaling_performance/optimizing_storage.adoc]: We should make that clear on the first usage. Always spell it out the first time.

File: scaling_performance/optimizing_storage.adoc
Comment (tmorriso-rh) [reply scaling_performance/optimizing_storage.adoc]: "GlusterFS provide interfaces..." ?

File: scaling_performance/optimizing_storage.adoc
Comment (tmorriso-rh) [reply scaling_performance/optimizing_storage.adoc]: Thin pool is a noun, and thin-pool is an adjective, correct?

**Comment by tmorriso-rh** [comment]
@nengard PTAL.

File: scaling_performance/optimizing_storage.adoc
Comment (nengard) [reply scaling_performance/optimizing_storage.adoc]: You are right, ignore my comment ;) hehe

**Comment by tmorriso-rh** [comment]
@nengard @lpsantil  I used bullet lists in the table to help with readability. I agree with Louis that the information is important since this a guidelines document. So, hoping the lists makes the table easier to read. PTAL.

**Comment by vikram-redhat** [comment]
@tmorriso-rh - Hey Traci - this hasn't been picked the 3.7 stage branch.

**Pull request merged**
